tests/pytorch/test_onnx_export.py

```diff
@@ -747,6 +747,28 @@ def test_export_layernorm_mlp(
         else:
             validate_result(fname, inp, model, atol=2e-2, is_fp8=use_fp8)
 
+def export_and_validate_flash_attn_models(
+        model: torch.nn.Module,
+        inp: Union[Tuple[torch.Tensor], torch.Tensor],
+        fname: str,
+        atol: float,
+        input_names: list=["input"],
+        export_fp8: bool=False,
+        validate_fp8: bool=False):
+    '''
+    Disabling usage of fused attention (flash_attention) as writing ONNX symbolic functions requires changes in the flash-attention library.
+    It is legal to do this because TE should behave the same when flash-attention is used and when it is not used.
+    We make sure to compare the results of TE with flash-attention to ONNX w/o flash attention, to ensure that this is legal.
+    '''
+    model.set_use_flash_attention(False)
```

Comment:
Hmm... How would the user actually export their model then? Do they need to set that before calling the export functionality? What about the cases where TE module is not the top level module (which would be majority of cases) - users would need to wire that setting in their model then?

Is there a way for us to somehow understand when the model is being exported to ONNX to use that information instead?

Reply:
This is a good point.
Nemo has done this kind of thing in the past. There was a setting (I don't know if it still exists) in Nemo or Megatron that changed some fusions to enable ONNX export.
@ptrendx the other alternative is to spend X amount of time now wiring FA for ONNX export which is costly in time and maintenance; or to integrate the FA code base into TE. We can discuss offline if this is not clear.

Reply:
Added context manager for onnx export, please review